tests/lib/e2e/buckets/files.e2e-spec.ts

```diff
@@ -0,0 +1,400 @@
+
+    axiosGetStub = sinon.stub(axios, 'get')
+    jest.clearAllMocks()
+    axiosGetStub.callsFake(async (url: string) => {
+      if (url.includes('/v2/upload/link')) return { data: { result: FAKE_UPLOAD_URL } }
+      if (url.includes('/v2/upload-multipart/link')) {
+        const parts = new URL(url).searchParams.get('parts')
+        return { data: { result: new Array(parts).fill(FAKE_UPLOAD_URL), UploadId: 'fake-id' } }
+      }
+      if (url.includes('/v2/download/link')) return { data: { result: FAKE_DOWNLOAD_URL } }
+      if (url.includes('exists')) return { status: 200 }
+    })
+  })
+
+  afterEach(() => {
+    axiosGetStub.restore()
+  })
+
+  describe('File Management v2', () => {
+
+    describe('Uploading a file', () => {
+
+      it('When a user wants to upload a file with just one part, it should work for owned buckets and get an upload link', async () => {
+        // Arrange: Create a bucket
+        const { body: { id: bucketId } } = await testServer
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+          .expect(201)
+
+        // Act: start the upload
+        const response = await testServer.post(`/v2/buckets/${bucketId}/files/start`)
+          .set('Authorization', getAuth(testUser))
+          .send({ uploads: [{ index: 0, size: 1000, }], })
+
+        // Assert
+        expect(response.status).toBe(200);
+        const { uploads } = response.body;
+
+        expect(uploads).toHaveLength(1);
+
+        const [upload] = uploads;
+
+        expect(upload.url).toBe(FAKE_UPLOAD_URL)
+        expect(upload.urls).toBeNull();
+        expect(upload.uuid).toBeDefined();
+
+      })
+
+      it('When a user wants to upload a file over 100MB with multiple parts into a owned bucket, it should work and get a list of upload links one per each file part', async () => {
+        // Arrange: Create a bucket
+        const { body: { id: bucketId } } = await testServer
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+          .expect(201)
+
+        // Act: start the upload
+        const MB100 = 100 * 1024 * 1024
+        const fileParts = [{ index: 0, size: MB100 / 2, }, { index: 1, size: MB100 / 2, },]
+        const response = await testServer.post(`/v2/buckets/${bucketId}/files/start?multiparts=${fileParts.length}`)
+          .set('Authorization', getAuth(testUser))
+          .send({ uploads: fileParts, })
+
+        // Assert
+        expect(response.status).toBe(200);
+        const { uploads } = response.body;
+
+        expect(uploads).toHaveLength(2);
+
+        const [firstUpload, secondUpload] = uploads;
+
+        expect(firstUpload.url).toBeNull();
+        expect(firstUpload.urls).toBeDefined();
+        expect(firstUpload.uuid).toBeDefined();
+        expect(firstUpload.UploadId).toBeDefined();
+
+        expect(secondUpload.url).toBeDefined();
+        expect(secondUpload.url).toBeNull();
+        expect(secondUpload.urls).toBeDefined();
```

Comment:
This check may be not enough, as we should check if the `urls` length matches with the parts number + the URLs inside the array are in fact, URLs.